include/fastrtps/rtps/common/Time_t.h

```diff
@@ -50,7 +50,22 @@ struct RTPS_DllAPI Time_t{
 		seconds = sec;
 		fraction = frac;
 	}
-	
+
+	Time_t(long double sec)
+	{
+		seconds = static_cast<uint32_t>(sec);
+		fraction = (sec - seconds) * 4294967296;
+	}
+
+	/**
+	 *  Returns stored time as nanoseconds
+	 */
+	inline int64_t to_ns()
+	{
+		int64_t nano = seconds * 1000000000;
+		nano += (fraction  * 1000000000) / 4294967296; // 1 fraction = 1/(2^32) seconds
```

Comment:
This will use 32 bit arithmetic. Please use ULL literals to all constants to force 64-bit arithmetic